registry/client_factory.go

```diff
@@ -43,47 +45,66 @@ func (f *RemoteClientFactory) ClientFor(repo image.CanonicalName, creds Credenti
 		tx = &logging{f.Logger, tx}
 	}
 
-	f.mx.Lock()
+	f.mu.Lock()
 	if f.challengeManager == nil {
 		f.challengeManager = challenge.NewSimpleManager()
 	}
-	f.mx.Unlock()
 	manager := f.challengeManager
+	f.mu.Unlock()
+
+	scheme := "https"
+	for _, h := range f.InsecureHosts {
```

Comment:
I can't not comment about this not being a map/set lookup, but it's incredibly unlikely to matter...

Reply:
Right -- I wouldn't expect enough entries here for O(n) to matter.